### DISCUSSION TITLE
registerCompletionItemProvider failed to trigger in jsx or tsx

### DISCUSSION THREAD

**Discussion by KangXinzhi** [microsoft/vscode-discussions#915] [Extension Development QnA] [answered]
When I use it in an HTML file, it can correctly trigger both letters and spaces within the class. But if in the JSX file, when I input letters inside the className="" attribute, it cannot be triggered, it can only be triggered through spaces.

example :
```
	let test3 = vscode.languages.registerCompletionItemProvider(
		'*',
		{
			provideCompletionItems(document, position) {
				const start = new vscode.Position(position.line, 0);
				const range = new vscode.Range(start, position);
				const text = document.getText(range);
				console.log('text',text)
				return new vscode.CompletionList([
					new vscode.CompletionItem('text-xl', vscode.CompletionItemKind.Value),
					new vscode.CompletionItem('text-gray-900', vscode.CompletionItemKind.Value),
					new vscode.CompletionItem('text-gray-800', vscode.CompletionItemKind.Value),
				]);
			},
		},
		'  ' // trigger space
	);
	context.subscriptions.push(test3);
	
```

<img width="649" alt="image" src="https://github.com/microsoft/vscode-discussions/assets/50164727/7e7edcd8-6a7e-4ebe-962f-a83fcf9be216">

<img width="418" alt="image" src="https://github.com/microsoft/vscode-discussions/assets/50164727/7ee6492c-373b-4c0b-bfed-c1827d28bd5d">

**Comment by tamuratak** [comment, marked as answer]
The difference might come from here. The built-in HTML extension sets `"` as a trigger:

https://github.com/microsoft/vscode/blob/c2abdd2738919a0e27d809a19bdce3005ad90046/extensions/html-language-features/server/src/htmlServer.ts#L197

```ts
            completionProvider: clientSnippetSupport ? { resolveProvider: true, triggerCharacters: ['.', ':', '<', '"', '=', '/'] } : undefined,
```

You had not better leave the trigger as a white space. What you should set as a trigger depends on the extension behavior you want.

**Reply by burkeholland** [reply]
Since this item is 3 weeks old with no update, we’re marking this as the correct answer for now. If this is not the case, please feel free to re-open and update.
